client/components/theme/index.jsx

```diff
@@ -172,8 +176,12 @@ const Theme = React.createClass( {
 								context: 'singular noun, the currently active theme'
 							} ) }</span>
 						}
-						{ price && ! purchased &&
-							<span className="theme-badge__price">{ price }</span>
+						{
+							config.isEnabled( 'jetpack/pijp' )
+							? ! this.props.isJetpack && price && ! purchased &&
```

Comment:
@eliorivero If we do decide to keep hiding the price on JP sites, let's do that in `themes-selection` by adding the relevant criteria to [`isThemePurchased`](https://github.com/Automattic/wp-calypso/blob/e9ddb66017c0a0361f11649710277a67eae275ba/client/my-sites/themes/themes-selection.jsx#L205) (and renaming it to `hidePremiumThemePrices`). That way, at least we don't pollute the `Theme` and `ThemesList` components with stuff that doesn't belong there.